agents/plugins/yum

```diff
@@ -1,125 +1,116 @@
-            # something is wrong here...
-            exit
-        fi
-                
-        # cache check results
-        # check if cached check result already exists and is nothing but a file
-        if [ -f $YUM_CACHE_FILE ] || [ ! -L $YUM_CACHE_FILE ]
-        then
-            echo $BOOT_REQUIRED > $YUM_CACHE_FILE
-            echo $UPDATES >> $YUM_CACHE_FILE
-            echo $SECURITY_UPDATES >> $YUM_CACHE_FILE
-        else
-            # something is wrong here...
-            echo "invalid check cache file"
-            exit
-        fi
+    count=0
+        while [ -n "$(pgrep -f "python /usr/bin/yum")" ]; do
+            if [ $count -eq 3 ]; then
+                echo "Tried to run yum for 30 secs but another yum instance was running"
+                exit 2
+            else
+                ((count++))
+                sleep 10
+            fi
+        done
+    LATEST_KERNEL=$(yum -q -C --noplugins --debuglevel 0 list installed | egrep "^(vz)?kernel(|-(uek|ml|lt))\." | grep "\." | tail -n1 | awk '{print $2};')
+    RUNNING_KERNEL=$(cat /proc/version | awk '{print $3}' | sed 's/.x86_64//g')
+    if [[ "$RUNNING_KERNEL" == "$LATEST_KERNEL"* ]]
+    then
+        BOOT_REQUIRED="no"
+    else
+        BOOT_REQUIRED="yes"
     fi
-else
-    if [ ! -s $YUM_CACHE_FILE ]
+    UPDATES=$(waitmax 25 /usr/bin/yum -C --noplugins --quiet list updates | grep "\." | cut -d' ' -f1 | wc -l || echo "-1")
+    # check if --security is available
+    waitmax 10 /usr/bin/yum -C --noplugins --quiet --security list updates > /dev/null 2>&1
+    if [ $? -eq 0 ]
     then
-        # fill cache file
-        echo $BOOT_REQUIRED
-        echo $UPDATES
-        echo $SECURITY_UPDATES
+        SECURITY_UPDATES=$(waitmax 25 /usr/bin/yum -C --noplugins --quiet --security list updates | grep "\." | cut -d' ' -f1 | wc -l || echo "-1")
     else
-        # use cache file
-        cat $YUM_CACHE_FILE
+    # --security not supported with this yum version
+    # maybe the yum-plugin-security package is needed (RH 6)
+            SECURITY_UPDATES="-2"
     fi
-fi
 
+    echo $BOOT_REQUIRED
+    echo $UPDATES
+    echo $SECURITY_UPDATES
+    # cache check yum
+    # check if cached check already exists and is nothing but a file
+    if [ -f $CACHE_YUM_UPDATE ] || [ ! -L $CACHE_YUM_UPDATE ]; then
+        echo "$YUM_CURRENT" > $CACHE_YUM_UPDATE
+    else
+        # something is wrong here...
+        echo "invalid check cache file"
+        exit 2
+    fi
+    # cache check results
+    # check if cached check result already exists and is nothing but a file
+    if [ -f $CACHE_RESULT_CHECK ] || [ ! -L $CACHE_RESULT_CHECK ]
+    then
+        echo $BOOT_REQUIRED > $CACHE_RESULT_CHECK
+        echo $UPDATES >> $CACHE_RESULT_CHECK
+        echo $SECURITY_UPDATES >> $CACHE_RESULT_CHECK
+    else
+        # something is wrong here...
+        echo "invalid check result cache file"
+        exit 2
+    fi
+else
+    # use cache file
+    cat $CACHE_RESULT_CHECK
+fi
```

Comment:
One last carriage return is missing - but I might add it as well :-)

Reply:
Ok ;-)